week3/src/components/Game.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useRef, useState } from "react";
+import GameItem from "./GameItem";
+import gameInfo from "../utils/constant";
+import { Header, Main } from "../style/style";
+import vs from "../assets/vs.png";
+import crown from "../assets/crown.png";
+
+function Game() {
+  const winners = useRef([]);
+  const round = useRef(gameInfo.length / 2);
+  const [fighters, setFighters] = useState(gameInfo.sort(() => Math.random() - 0.5));
+  const [leftRound, setLeftRound] = useState(round.current);
+  const [win, setWin] = useState(false);
+
+  const handleClick = (fighter) => {
+    winners.current.push(fighter);
+    setLeftRound((current) => current - 1);
+    setFighters(fighters.slice(2));
+
+    if (fighters.length === 2) {
+      if (winners.current.length === 1) {
+        setFighters([]);
+        setWin(true);
+        return;
+      }
+      setFighters(winners.current);
+      winners.current = [];
+      round.current /= 2;
+      setLeftRound(round.current);
+    }
```

Comment:
```suggestion
      if (fighters.length !== 2) return;
      if (winners.current.length === 1) {
        setFighters([]);
        setWin(true);
        return;
      }
      setFighters(winners.current);
      winners.current = [];
      round.current /= 2;
      setLeftRound(round.current);
```

이 부분 이런 식으로 수정하면 중첩 if문 탈출할 수 있어서 좀 더 깔끔할 것 같아여 헤헤... 근데 이것도 클린코드영역이라 갠취 반영임다 ㅋ ㅋ

Reply:
생각지도 못한 걸 많이 배워갑니다.. 👍 사실 이중 반복문 쓰면서도 이것밖에 방법이 없나 생각을 많이 했는데 제시해주신 방법이 더 깔끔한 것 같아서 좋네요!

Reply:
우와...... 지나가다 진짜 이마탁 하고 지나갑니다.....